src/main/java/com/automatedtest/sample/driver/TearDown.java

```diff
@@ -11,15 +11,15 @@
     private WebDriver driver;
 
     public TearDown() {
-        this.driver = Setup.driver;
+        this.driver = Setup.currentDriver.get();
     }
 
     @After
     public void quitDriver(Scenario scenario) {
         if (scenario.isFailed()) {
             saveScreenshotsForScenario(scenario);
         }
-        this.driver.quit();
+        Setup.currentDriver.get().quit();
```

Comment:
I don't think you have change this. it's already set to the correct driver in the constructor

Reply:
yes, I reverted code.